### PR TITLE
[do not merge] update samples to take Microsoft Identity Web as a nuget package

### PR DESCRIPTION
**Pull request by jennyf19**
- I have tested all the samples, but would appreciate another set or two of eyes. Not sure if I've caught all the ReadMe updates. Appreciate any testing you can do as well. thanks.

### CODE REVIEW

File: 5-WebApp-AuthZ/5-2-Groups/Startup.cs [hunk 1]
```diff
@@ -36,16 +36,13 @@ public void ConfigureServices(IServiceCollection services)
...
+            services.AddSignIn(options =>
                 {
-                    Configuration.Bind("AzureAD", options);
-
                     // Uncomment the following lines code instruct the asp.net core middleware to use the data in the "groups" claim in the [Authorize] attribute and for User.IsInrole()
                     // See https://docs.microsoft.com/en-us/aspnet/core/security/authorization/roles for more info.
                     //    // Use the groups claim for populating roles
                     //    options.TokenValidationParameters.RoleClaimType = "groups";
-                });
+                }, options => { });
```
Comment (jmprieur) [inline 5-WebApp-AuthZ/5-2-Groups/Startup.cs:45]: the second options should probably do the binding as well? `Configuration.Bind("AzureAD", options);` ?

File: 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs [hunk 2]
```diff
@@ -106,7 +85,7 @@ public void ConfigureServices(IServiceCollection services)
 
                         return Task.FromResult(0);
                     };
-                });
+                }, option => { });
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:88]: the second options callback should probably do the binding as well? `Configuration.Bind("AzureAD", options);` ?

File: 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs [hunk 3]
```diff
@@ -78,7 +57,7 @@ public void ConfigureServices(IServiceCollection services)
 
             // Sign-in users with the Microsoft identity platform
             services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn("AzureAD", Configuration, options =>
+                .AddSignIn(options =>
                 {
                     Configuration.Bind("AzureAD", options);
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:63]: We also need to check what happen here. I'm not sure how the overrides of the events will be handled

File: 5-WebApp-AuthZ/5-2-Groups/WebApp-OpenIDConnect-DotNet.csproj [hunk 4]
```diff
@@ -19,11 +19,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.Graph" Version="1.21.0" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <ProjectReference Include="..\..\Microsoft.Identity.Web.UI\Microsoft.Identity.Web.UI.csproj" />
-    <ProjectReference Include="..\..\Microsoft.Identity.Web\Microsoft.Identity.Web.csproj" />
+    <PackageReference Include="Microsoft.Identity.Web.UI" Version="0.1.0-localbuild" />
```
Comment (jmprieur) [inline 5-WebApp-AuthZ/5-2-Groups/WebApp-OpenIDConnect-DotNet.csproj:22]: Oh, I understand Microsoft.Identityu.Web.UI also references Microsoft.Identity.Web ...

File: 1-WebApp-OIDC/1-1-MyOrg/Startup.cs [hunk 5]
```diff
@@ -35,8 +35,8 @@ public void ConfigureServices(IServiceCollection services)
 
             // Sign-in users with the Microsoft identity platform
             //services.AddSignIn(Configuration);
-            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn("AzureAD", Configuration, options => Configuration.Bind("AzureAD", options));
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme);
```
[the remark this answers is not in the document]
Comment (jennyf19) [reply 1-WebApp-OIDC/1-1-MyOrg/Startup.cs:38]: thanks. had fixed this but the changes didn't get pushed up last night. will do. :)

File: 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs [hunk 6]
```diff
@@ -56,11 +56,8 @@ public void ConfigureServices(IServiceCollection services)
             services.AddScoped<IMSGraphService, MSGraphService>();
 
             // Sign-in users with the Microsoft identity platform
-            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn(options =>
+            services.AddSignIn(options =>
                 {
-                    Configuration.Bind("AzureAD", options);
```
Comment (jmprieur) [inline 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:83]: I think we still need `Configuration.Bind("AzureAD", options);` both here and in line 88.

File: 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs (hunk 3 above)
Comment (jennyf19) [reply 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs]: we can sync on this

---
In reply to: [406347523](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406347523) [](ancestors = 406347523)

File: 5-WebApp-AuthZ/5-2-Groups/WebApp-OpenIDConnect-DotNet.csproj (hunk 4 above)
Comment (jennyf19) [reply 5-WebApp-AuthZ/5-2-Groups/WebApp-OpenIDConnect-DotNet.csproj:23]: fixing these. thanks.

---
In reply to: [406348152](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406348152) [](ancestors = 406348152)

**Comment by jennyf19** [comment]
i think addressed all the comments. thank you.

---
In reply to: [390969734](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#issuecomment-390969734) [](ancestors = 390969734)

File: 1-WebApp-OIDC/1-1-MyOrg/WebApp-OpenIDConnect-DotNet.csproj [hunk 7]
```diff
@@ -25,12 +25,9 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.0.0" />
+    <PackageReference Include="Microsoft.Identity.Web" Version="0.1.0-localbuild" />
```
Comment (pmaytak) [inline 1-WebApp-OIDC/1-1-MyOrg/WebApp-OpenIDConnect-DotNet.csproj:28]: Hmm, should version be `0.1.0-preview` or something?

File: 5-WebApp-AuthZ/5-2-Groups/Startup.cs [hunk 8]
```diff
@@ -36,15 +36,15 @@ public void ConfigureServices(IServiceCollection services)
             });
 
             // Sign-in users with the Microsoft identity platform
-            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn("AzureAD", Configuration, options =>
+            services.AddSignIn(options =>
```
Comment (pmaytak) [inline 5-WebApp-AuthZ/5-2-Groups/Startup.cs:39]: Missing `Configuration.Bind("AzureAD", options);` in the first options action?

File: 1-WebApp-OIDC/1-1-MyOrg/WebApp-OpenIDConnect-DotNet.csproj (hunk 7 above)
Comment (jennyf19) [reply 1-WebApp-OIDC/1-1-MyOrg/WebApp-OpenIDConnect-DotNet.csproj:28]: I know. I wanted to test everything w/the nuget package w/the current release build. we will have to run a new package w/the correct name.

**Comment by jennyf19** [comment]
@TiagoBrenck I ran the [build again](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=508771&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=368b987b-4e63-5984-5006-1d171be4473f) and things are passing. did you address all the comments above in your earlier PR? thx.

File: 1-WebApp-OIDC/1-1-MyOrg/README.md [hunk 9]
```diff
@@ -164,8 +164,7 @@ cd "1-WebApp-OIDC\1-1-MyOrg"
      by this line:
 
      ```CSharp
-      services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-              .AddSignIn("AzureAd", Configuration, options => Configuration.Bind("AzureAd", options));
+      services.AddSignIn(Configuration);
```
[the remark this answers is not in the document]
Comment (jennyf19) [reply 1-WebApp-OIDC/1-1-MyOrg/README.md:167]: yes, we probably should. can you do that?

---
In reply to: [406857441](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406857441) [](ancestors = 406857441)

File: 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs (hunk 6 above)
Comment (jennyf19) [reply 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:83]: I've fixed it. thx

---
In reply to: [406889283](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406889283) [](ancestors = 406889283)

**Comment by jennyf19** [comment]
this was addressed right?

---
In reply to: [391557054](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#issuecomment-391557054) [](ancestors = 391557054)

**Comment by jennyf19** [comment]
@jmprieur this sample is ready to merge. we will let you do a final pass and merge when ready. @TiagoBrenck  has a few points to discuss and will do so via email. :)

**Pull request merged**